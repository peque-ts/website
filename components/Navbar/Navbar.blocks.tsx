import Image from 'next/image';
import Link from 'next/link';
import React from 'react';

import { NAV_ITEMS } from './Navbar.helpers';

interface IconLinkProps {
  icon: string;
  link: string;
  alt: string;
}

const IconLink: React.VFC<IconLinkProps> = ({ link, icon, alt }) => (
  <Link href={link}>
    <a>
      <Image src={icon} alt={alt} width={40} height={40} />
    </a>
  </Link>
);

const Nav: React.VFC = () => (
  <nav>
    <ul className="flex space-x-8">
      {NAV_ITEMS.map(({ name, link }, index) => (
        <li key={index}>
          <Link href={link}>
            <a className="text-lg">{name}</a>
          </Link>
        </li>
      ))}
    </ul>
  </nav>
);

export { IconLink, Nav };
