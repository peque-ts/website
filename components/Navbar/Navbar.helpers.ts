interface NavItem {
  name: string;
  link: string;
}

export const NAV_ITEMS: NavItem[] = [
  {
    name: 'Docs',
    link: '/docs',
  },
  {
    name: 'Tutorials',
    link: '/tutorials',
  },
  {
    name: 'Enterprise',
    link: '/enterprise',
  },
  {
    name: 'Sponsors',
    link: '/sponsors',
  },
];
