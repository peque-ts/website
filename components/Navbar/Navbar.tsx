import React from 'react';

export const Navbar: React.VFC = () => {
  return <div>navbar</div>;
};
