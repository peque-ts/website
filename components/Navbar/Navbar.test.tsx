import { render, screen } from '@testing-library/react';

import { Navbar } from './Navbar';

describe('Navbar', () => {
  it('should render', () => {
    const { asFragment } = render(<Navbar />);

    expect(screen.getByRole('navigation')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
