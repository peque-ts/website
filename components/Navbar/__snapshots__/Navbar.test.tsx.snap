// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render 1`] = `
<DocumentFragment>
  <div>
    navbar
  </div>
</DocumentFragment>
`;
